backend/src/app/core/data/crud/source_document_metadata.py

```diff
@@ -3,16 +3,36 @@
 from loguru import logger
 from sqlalchemy.orm import Session
 
-from app.core.data.crud.crud_base import CRUDBase, NoSuchElementError
+from app.core.data.crud.crud_base import CRUDBase
 from app.core.data.dto.action import ActionType
+from app.core.data.dto.project_metadata import ProjectMetadataRead
 from app.core.data.dto.source_document_metadata import (
+    SourceDocumentMetadataBaseDTO,
     SourceDocumentMetadataCreate,
     SourceDocumentMetadataUpdate,
 )
-from app.core.data.orm.source_document import SourceDocumentORM
+from app.core.data.meta_type import MetaType
+from app.core.data.orm.project_metadata import ProjectMetadataORM
 from app.core.data.orm.source_document_metadata import SourceDocumentMetadataORM
 
 
+# TODO: wohin damit?
```

Comment:
You could create a [pydantic field validator](https://docs.pydantic.dev/latest/concepts/validators/#model-validators) on the DTO directly. That way, fastapi automatically handles validation :)